pkg/controller/metrics/provision_underway_collector.go

```diff
@@ -164,6 +175,15 @@ func (cc provisioningUnderwayInstallRestartsCollector) Collect(ch chan<- prometh
 			continue // skip reporting the metric for clusterdeployment until the InstallRestarts is at least minRestarts
 		}
 
+		// Skip logging the metric if conditions in desired state
```

Comment:
I would rather see all the logic embedded in `getKnownConditions`, so you can immediately (after L162) do something like:
```go
if condition == something {
	continue
}
```